src/include/mallocMC/creationPolicies/Scatter_impl.hpp

```diff
@@ -1019,10 +1024,15 @@ namespace mallocMC
                     typename alpaka::dim::traits::DimType<AlpakaAcc>::type;
                 using Idx =
                     typename alpaka::idx::traits::IdxType<AlpakaAcc>::type;
+                using VecType = alpaka::vec::Vec<Dim, Idx>;
+
+                auto threadsPerBlock = VecType::ones();
+                threadsPerBlock[Dim::value - 1] = 256u;
+
                 const auto workDiv = alpaka::workdiv::WorkDivMembers<Dim, Idx>{
-                    Idx{1},
-                    Idx{256},
-                    Idx{1}}; // Dim may be any dimension, but workDiv is 1D
+                    VecType::ones(),
+                    threadsPerBlock,
+                    VecType::ones()}; // Dim may be any dimension, but workDiv is 1D
```

Comment:
What is the purpose of this change? The original version should have the same behavior and is shorter.

Reply:
The original version produces compile time errors because of https://github.com/alpaka-group/alpaka/issues/1035
The changes setting always the x component (highest) index depending of the user provided accelerator.

Reply:
Hmm. But that did not happen with the alpaka subtree? Then we should probably also update the alpaka subtree. Did we not release an alpaka version recently? We could update the subtree to this version.

As for the indexing, that once worked independently on the dimension of the accelerator. But maybe this was more of a bug than a feature. Anyway, if it works now as well, I am fine with it!

Reply:
It was with the alpaka subtree and if I remember correctly with HIP-clang (clang 10).
I will keep it for now as it is, because it is save and open a issue that we invest later time into it.